### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by mrprotocoll**
### The above updates were made

- Setup react and redux
- Setup Linters
- Add required packages
- Setup routers
- Setup redux store
- Add Home and Details routes
- Add StockItem component
- Add stock Details components
- Add hero component
- Add stock data to store
- Displayed hero and stock on the homepage
- Add styling for the home and details page
- Add unit testing
- Add integration testing
- Update README

### CODE REVIEW

File: src/components/Home.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  return (
+    <div className="container">
+      <Hero data={hero} />
+      <ul className={styles.items}>
+        {
+          stocks.map((stock) => <StockItem key={stock.symbol} data={stock} />)
+        }
+      </ul>
+    </div>
```
Comment (elisha2kyakpo1) [inline src/components/Home.js]: - [ ] You have done a great job, especially on the design part, but you may have missed an important requirement for this project. We are told to create a search functionality where users would search for an item using it's name, I have attached an image for your reference.

search

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @mrprotocoll, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Home.js (hunk 1 above)
Comment (mrprotocoll) [reply src/components/Home.js]: Thank you @elisha2kyakpo1 
I didn't think that was part of the requirement as it was not stated under the project requirement, I will add it and request a new review.

File: src/components/Home.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  }, [dispatch]);
+
+  return (
+    <div className="container">
+      <Hero data={hero} />
+      <ul className={styles.items}>
+        {
+          stocks.map((stock) => <StockItem key={stock.symbol} data={stock} />)
+        }
+      </ul>
```
Comment (bobb-Rob) [inline src/components/Home.js]: - [ ] As mentioned by the previous reviewer, you are expected to implement a filter on the home page. See image of the requirement below 👇🏾 

image

- When the page loads, your webapp should show the list of categories that could be **_filtered out by some parameter, for example by the category name._** 
Kindly implement this.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️

Hi @mrprotocoll, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Home.js (hunk 2 above)
Comment (mrprotocoll) [reply src/components/Home.js]: Thank you @bobb-Rob , i will make the changes and request a new review

**Review by jerryowusu — APPROVED** [review]
Hi @mrprotocoll , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

chelsea-mason-approve


## Highlights
No linter errors :heavy_check_mark: 
Great job so far :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**